lib/php/rendition-factory/src/Transformer/Video/VideoTransformerBase.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Alchemy\RenditionFactory\Transformer\Video;
+
+use Alchemy\RenditionFactory\Config\ModuleOptionsResolver;
+use Alchemy\RenditionFactory\Context\TransformationContextInterface;
+use Alchemy\RenditionFactory\Transformer\Video\FFMpeg\Format\FormatInterface;
+use FFMpeg;
+use Symfony\Component\DependencyInjection\Attribute\AutowireLocator;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+
+readonly class VideoTransformerBase
+{
+    protected ?string $format;
+    protected FormatInterface $outputFormat;
+    protected ?string $extension;
+    protected TransformationContextInterface $context;
+    protected FFMpeg\FFMpeg $ffmpeg;
+
+    public function __construct(#[AutowireLocator(FormatInterface::TAG, defaultIndexMethod: 'getFormat')] protected ServiceLocator $formats,
+        protected ModuleOptionsResolver $optionsResolver,
+    ) {
+    }
+
+    protected function prepare(array $options, TransformationContextInterface $context): void
+    {
+        $this->context = $context;
```

Comment:
It should not be stored.
(Also you would have a "readonly" error at the next pass)